packages/flutter/lib/src/widgets/web_image.dart

```diff
@@ -0,0 +1,68 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../../widgets.dart';
+import '_web_image_io.dart' if (dart.library.js_interop) '_web_image_web.dart';
+
+/// A widget that displays an image from the web.
+///
+/// Due to restrictions on cross-origin images on the web, in cases where the
+/// image [url] is cross-origin and access to the byte data is disallowed, this
+/// widget will instead use a platform view to display the image in an `<img>`
+/// element.
+///
+/// If the [src] of the image is same-origin, or if we are able to make a
+/// cross-origin request for the image bytes, then this widget works exactly
+/// the same as [Image.network].
+class WebImage extends StatefulWidget {
+  /// Creates a web image.
+  const WebImage(
+    this.src, {
+    super.key,
+    this.loadingBuilder,
+    this.frameBuilder,
+    this.errorBuilder,
+  });
+
+  final String src;
+  final ImageLoadingBuilder? loadingBuilder;
+  final ImageFrameBuilder? frameBuilder;
+  final ImageErrorWidgetBuilder? errorBuilder;
+
+  @override
+  State<WebImage> createState() => _WebImageState();
+}
+
+class _WebImageState extends State<WebImage> {
+  bool checkedIfBytesCanBeFetched = false;
+  bool imageBytesCanBeFetched = true;
+  @override
+  void initState() {
+    super.initState();
+    // Check if `widget.src` can be fetched.
+    checkIfImageBytesCanBeFetched(widget.src).then((bool canBeFetched) {
```

Comment:
I used to use a GET request to check if the bytes were fetchable before, which would cause the cache to be used in the second request to actually render the `Image.network`. The downside of the GET approach is that it will take longer to determine if the bytes are fetchable because we will have to wait for the browser to actually fetch the bytes. Then later when the same `src` is used in `Image.network`, the request should be cached and the image ready to display pretty much immediately.